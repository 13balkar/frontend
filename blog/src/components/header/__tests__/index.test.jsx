import { render, screen } from '@testing-library/react';
import * as React from 'react';
import Header from '..';
describe('Header', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly with header data', () => {
    render(<Header/>);
    expect(screen.getByText('The Artifact')).toBeInTheDocument();
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(3);
    expect(links[0]).toHaveTextContent('Blog');
    expect(links[1]).toHaveTextContent('About');
    expect(links[2]).toHaveTextContent('Contact');
    expect(links[0]).toHaveAttribute('href', '');
    expect(links[1]).toHaveAttribute('href', '');
    expect(links[2]).toHaveAttribute('href', '');
  });
});
