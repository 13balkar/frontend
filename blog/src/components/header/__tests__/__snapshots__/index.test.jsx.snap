// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<DocumentFragment>
  <header
    class="outerPadding"
  >
    <div
      class="headerName"
    >
      <h2>
        The Artifact
      </h2>
      <h3>
        <i>
          Culture and Art Blog
        </i>
      </h3>
    </div>
    <nav
      class="headerNav"
    >
      <a
        href=""
      >
        Blog
      </a>
      <a
        href=""
      >
        About
      </a>
      <a
        href=""
      >
        Contact
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
